jerry-core/jerry-api.h

```diff
@@ -147,71 +173,83 @@ jerry_value_t jerry_value_to_string (const jerry_value_t);
 jerry_value_t jerry_value_remove_error_flag (const jerry_value_t);
 
 /**
- * Create functions of 'jerry_value_t'
- */
-jerry_value_t jerry_create_null_value (void);
-jerry_value_t jerry_create_undefined_value (void);
-jerry_value_t jerry_create_boolean_value (bool);
-jerry_value_t jerry_create_number_value (double);
-jerry_value_t jerry_create_object_value (jerry_object_t *);
-jerry_value_t jerry_create_string_value (jerry_string_t *);
-
-/**
  * Acquire types with reference counter (increase the references)
  */
-jerry_string_t *jerry_acquire_string (jerry_string_t *);
-jerry_object_t *jerry_acquire_object (jerry_object_t *);
 jerry_value_t jerry_acquire_value (jerry_value_t);
 
 /**
  * Relase the referenced values
  */
-void jerry_release_object (jerry_object_t *);
-void jerry_release_string (jerry_string_t *);
 void jerry_release_value (jerry_value_t);
 
 /**
- * Create functions of API objects
+ * Create functions of API values
  */
-jerry_object_t *jerry_create_object (void);
-jerry_object_t *jerry_create_array_object (jerry_size_t);
-jerry_object_t *jerry_create_external_function (jerry_external_handler_t);
-jerry_object_t *jerry_create_error (jerry_error_t, const jerry_char_t *);
-jerry_object_t *jerry_create_error_sz (jerry_error_t, const jerry_char_t *, jerry_size_t);
-jerry_string_t *jerry_create_string (const jerry_char_t *);
-jerry_string_t *jerry_create_string_sz (const jerry_char_t *, jerry_size_t);
+jerry_value_t jerry_create_array (jerry_size_t);
+jerry_value_t jerry_create_boolean (bool);
+jerry_value_t jerry_create_error (jerry_error_t, const jerry_char_t *);
+jerry_value_t jerry_create_error_sz (jerry_error_t, const jerry_char_t *, jerry_size_t);
+jerry_value_t jerry_create_external_function (jerry_external_handler_t);
+jerry_value_t jerry_create_number (double);
+jerry_value_t jerry_create_null (void);
+jerry_value_t jerry_create_object (void);
+jerry_value_t jerry_create_string (const jerry_char_t *);
+jerry_value_t jerry_create_string_sz (const jerry_char_t *, jerry_size_t);
+jerry_value_t jerry_create_undefined (void);
 
 /**
  * Functions of array objects
  */
-bool jerry_set_array_index_value (jerry_object_t *, jerry_length_t, jerry_value_t);
-bool jerry_get_array_index_value (jerry_object_t *, jerry_length_t, jerry_value_t *);
-uint32_t jerry_get_array_length (const jerry_object_t *);
+jerry_value_t jerry_set_array_index_value (const jerry_value_t, uint32_t, const jerry_value_t);
```

Comment:
Currently it checks the argument, but we can make it more general. What about 'jerry_set_value_at_index'?